### PR TITLE
Fixes message complexity and simplifies implementation.

### PR DESCRIPTION
**Pull request by pontuslae**

### CODE REVIEW

File: Application/src/view/visuals/component/Message.java [hunk 1]
```diff
@@ -90,76 +84,81 @@ public boolean isKeepAnimating() {
...
+
+                // Sending a return message.
+            } else if(node1.getX() > node2.getX()){
+
+                if((animationBounds -= move) < (this.node2.getX())  - this.node1.getX()) {
+                    animating = false;
+                }
+
+                // Self calls
+            } else /*if (node1.getX() == node2.getX())*/ {
```
Comment (IsabelleTornqvist) [inline Application/src/view/visuals/component/Message.java:110]: ew remove pls

File: Application/src/view/visuals/component/Message.java [hunk 2]
```diff
@@ -20,31 +20,25 @@
     private int offset; // offset for the message "ordering".
     private int class_size; // identifier for the current class size so that messages can scale.
     private int animationBounds = 0; //Starts and stops the animation at the correct bounds.
-    private boolean keepAnimating = true; // State of the animation. Beginning or ending.
+    private boolean animating = true; // State of the animation. Beginning or ending.
     private boolean switchImage; // Keeps track of which image to show.
     private double messageScale = 1.5;
     private int selfCallCounter = 1; // Used for the hardcoded trajectory of self referencing message.
-    private boolean switchDirection = false; // Keeps track whether a the dragon's flying direction should be switched
```
Comment (kosara-g) [inline Application/src/view/visuals/component/Message.java:27]: how's the flying direction handled now?

File: Application/src/view/visuals/component/Message.java [hunk 3]
```diff
@@ -90,76 +84,81 @@ public boolean isKeepAnimating() {
...
+                    default :
+                        selfCallCounter = 1;
+                }
+            }
+        }
 
-               // Resets static indicator.
-               staticIndicator = false;
+        // If it's being viewed in the past and not currently animating.
+        if (staticIndicator) {
```
Comment (kosara-g) [inline Application/src/view/visuals/component/Message.java:143]: I still think this variable has a vague name; also static means something else in Java

File: Application/src/view/visuals/component/Message.java [hunk 4]
```diff
@@ -262,78 +263,33 @@ public void renderDefault(GraphicsContext gc) {
...
+            gc.drawImage(dragon, x1 + animationBounds, y1 + (this.class_size), width, height);
 
-            //Checks if down image is supposed to be shown. if this one is used we are sending a return message.
-            else{
-                //sets the dimensions of the dragon according to the current class size.
-                gc.drawImage(dragonMessageRev2, x1 + animationBounds,
-                        y1 + (this.class_size), class_size/messageScale, class_size/messageScale); //State Wings Down.
-                switchImage = true;
-            }
+            // Swaps the image.
```
Comment (kosara-g) [inline Application/src/view/visuals/component/Message.java:287]: which image?

File: Application/src/view/visuals/component/Message.java (hunk 2 above)
Comment (pontuslaestadius) [reply Application/src/view/visuals/component/Message.java:27]: I removed a redundant variable,
directionSwitched is used now.

Comment (kosara-g) [reply Application/src/view/visuals/component/Message.java:27]: +1

File: Application/src/view/visuals/component/Message.java (hunk 4 above)
Comment (Sebastian-Fransson) [reply Application/src/view/visuals/component/Message.java:287]: +1

File: Application/src/view/visuals/component/Message.java (hunk 3 above)
Comment (Sebastian-Fransson) [reply Application/src/view/visuals/component/Message.java:143]: +1

File: Application/src/view/visuals/component/Message.java (hunk 1 above)
Comment (IsabelleTornqvist) [reply Application/src/view/visuals/component/Message.java:110]: /*if (node1.getX() == node2.getX())*/

File: Application/src/view/visuals/component/Message.java (hunk 3 above)
Comment (pontuslaestadius) [reply Application/src/view/visuals/component/Message.java:143]: This was added 3 sprints ago.

Comment (pontuslaestadius) [reply Application/src/view/visuals/component/Message.java:143]: It was not an issue then?

Comment (kosara-g) [reply Application/src/view/visuals/component/Message.java:143]: if we're improving the code, might as well change it now is all

**Comment by pontuslaestadius** [comment]
Included in Demo

**Pull request closed without merging**